# **Analyse de la base de données du titanic**

**Notre objectif** dans l'analyse de cette base de données est de prédire si un passager aurait survécu au nauvrage du *titanic*
Avec une **métrique** de 90%. 

## **Analyse exploratoire de la base de données du titanic**

In [1]:
import numpy as np
import pandas as pd 
import seaborn as sns
import matplotlib.pyplot as plt

### **Chargement des packages**

In [2]:
data = pd.read_csv("../input/titanic/train.csv", index_col = "PassengerId")

### **Description de la base de données**

In [3]:
data.head()

,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
PassengerId,,,,,,,,,,,
1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [4]:
data.shape

(891, 11)

Notre base de données possède 891 observations et 11 variables.

survival	Survie	0 = No, 1 = Yes
pclass	classes de billet	1 = 1st, 2 = 2nd, 3 = 3rd
sex 	 Sexe
Age 	Age en année
sibsp	# de frères et sœurs / conjoints à bord du Titanic 
parch	# de parents/enfants à bord du Titanic
ticket	Numéro de billet
fare	Tarif passager
cabin	Numéro de cabine
embarked	Port d'Embarquation	C = Cherbourg, Q = Queenstown, S = Southampton

**Target variable** : notre variable à prédire ici c'est : *Survived*

In [5]:
missing_values_columns = ((data.isnull().sum()/data.shape[0])*100).sort_values(ascending=False)

In [6]:
pourcent_missing_values = (missing_values_columns.sum()/np.product(data.shape))*100

In [7]:
pourcent_missing_values

0.9916759908257374